src/dom0.c

```diff
@@ -23,13 +28,22 @@ int domain_get_user_cfg_count(void)
 struct xen_domain_cfg *domain_get_user_cfg(int index)
 {
 	if (index < domain_get_user_cfg_count()) {
-		return domain_cfgs[index];
+		return domain_cfgs[index].domain_cfg;
 	}
 
 	return NULL;
 }
 
 int main(void)
 {
-	return 0;
+	int i = 0;
+
+	/* init domains cfg */
+	while (domain_cfgs[i].domain_cfg) {
+		domain_cfgs[i].domain_cfg->image_info = &domain_cfgs[i];
```

Comment:
I don't understand what is going here. Why are you initializing `image_info` with pointer to `domain_cfgs` itself?

Reply:
So it can be passed back in image callback. There is no other option to pass application data there. I'll add comment